src/main/java/com/google/devtools/build/lib/analysis/config/HostTransition.java

```diff
@@ -50,4 +53,37 @@ public BuildOptions patch(BuildOptions options) {
       return options.createHostOptions();
     }
   }
+
+  /** Returns a {@link TransitionFactory} instance that generates the host transition. */
+  public static <T extends TransitionFactoryData> TransitionFactory<T> createFactory() {
+    return new AutoValue_HostTransition_Factory<>();
+  }
+
+  /**
+   * Returns {@code true} if the given {@link TransitionFactory} is an instance of the host
+   * transition.
+   */
+  public static <T extends TransitionFactoryData> boolean isInstance(
```

Comment:
Is this only for testing?

Reply:
No, it helps to replace all of the existing "transition.isHost()" style checks.